### PR TITLE
Fix CI Build: avoid test on windows, remove Comiso module, specific xcode on github

### LINKED ISSUE #2363: ray_mesh_intersect: corner-case fails on windows CI update
One of these tests just started failing on Windows. I'm not sure what changed on the CI machines between:

❌ https://github.com/libigl/libigl/actions/runs/8147735595/job/22270837016
✅ https://github.com/libigl/libigl/actions/runs/7976689950/job/21777837779

The code for libigl is identical, so something in the compilation. I'm not a windows user so this is really difficult for me to debug. @eolivi-fy  or @jnordin20 would you be able to help me ?

_Originally posted by @alecjacobson in https://github.com/libigl/libigl/issues/2271#issuecomment-1977726806_

### PR DESCRIPTION
**Pull request by alecjacobson**
Fixes #2363

### CODE REVIEW

File: tests/include/igl/ray_mesh_intersect.cpp [hunk 1]
```diff
@@ -30,6 +30,8 @@ TEST_CASE("ray_mesh_intersect: one_triangle", "[igl]")
   IGL_POP_FPE;
 }
 
+// https://github.com/libigl/libigl/issues/2363
+#ifndef _WIN32
```
Comment (jdumas) [inline tests/include/igl/ray_mesh_intersect.cpp:34]: Rather than not compile the test on Windows, I would mark it as `[!mayfail]` so it is compiled but not run (and can easily be run manually). Something like this:

```c++
#ifdef _WIN32
#define IGL_MAYFAIL_WIN "[!mayfail]"
#else
#define IGL_MAYFAIL_WIN
#endif

TEST_CASE("ray_mesh_intersect: corner-case", "[igl]" IGL_MAYFAIL_WIN)
{
...
}
```

Comment (alecjacobson) [reply tests/include/igl/ray_mesh_intersect.cpp:34]: oh nice! definitely let's do that.

**Comment by alecjacobson** [comment]
Fixes #2383 too (hopefully)

**Comment by alecjacobson** [comment]
I'm going to take this moment to remove the comiso module. It was put on notice with the last release. I will not be maintaining it. If someone else would like to, then happy to accept a PR adding it back.
